backend/api/signals.py

```diff
@@ -29,6 +29,15 @@ def handle_stock_post_save(instance: models.Stock, created: bool, **kwargs):
                 ),
             )
 
+        models.Stock.objects.update_or_create(
+            ticker=instance.ticker,
+            defaults=dict(
+                name=instance.name,
+                summary=instance.summary,
+                image_url=services.get_image_url(instance.name),
+            ),
+        )
+
```

Comment:
If you just want to add the image url, it can be done together with the object initialisation since the object is created in the services. Doing this is potentially unsafe as this Signal runs when the save method is called on a stock object (we check if its created or not)